### PR TITLE
[WIP] Augmentation benchmark

### PR DESCRIPTION
**Pull request by robsdavis**
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Affected Dependencies
None

## How has this been tested?
- [ ] Tests added

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/vanderschaarlab/.github/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/vanderschaarlab/.github/blob/main/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [van der Schaar Lab Styleguide](https://github.com/vanderschaarlab/.github/blob/main/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/vanderschaarlab/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: src/synthcity/metrics/eval.py [hunk 1]
```diff
@@ -175,17 +185,32 @@ def evaluate(
...
+                    metric(
+                        reduction=reduction,
+                        n_histogram_bins=n_histogram_bins,
+                        task_type=task_type,
+                        random_state=random_state,
+                        workspace=workspace,
+                        use_cache=use_cache,
+                    ),
+                    X_gt.sample(eval_cnt),
+                    X_augmented.sample(eval_cnt),
```
Comment (bcebere) [inline src/synthcity/metrics/eval.py:199]: Is `sample(eval_cnt)` relevant for the augmented dataset? The augmented dataset will be larger than `X_gt` everytime, right?

File: tests/plugins/survival_analysis/test_survival_ctgan.py [hunk 2]
```diff
@@ -65,7 +65,7 @@ def test_plugin_hyperparams(test_plugin: Plugin) -> None:
     ],
 )
 def test_plugin_fit(tte_strategy: str) -> None:
-    test_plugin = plugin(tte_strategy=tte_strategy, device="cpu", **plugins_args)
+    test_plugin = plugin(tte_strategy=tte_strategy, device="zz", **plugins_args)
```
[the remark this answers is not in the document]
Comment (robsdavis) [reply tests/plugins/survival_analysis/test_survival_ctgan.py]: Typo. Now fixed

File: src/synthcity/benchmark/utils.py [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+        # use user-specified values to augment
+        if not ad_hoc_augment_vals:
+            raise ValueError(
+                "When augmenting with an `ad-hoc` method, ad_hoc_augment_vals must be a dictionary, where the dictionary keys are the values of the fairness_column and the dictionary values are the number of records to augment."
+            )
+        else:
+            if not set(ad_hoc_augment_vals.keys()).issubset(
+                set(X_train[fairness_column].values)
+            ):
+                print(set(X_train[fairness_column].values))
```
Comment (bcebere) [inline src/synthcity/benchmark/utils.py:72]: Don't leave prints in the code. use `log` if the logs are needed.

File: src/synthcity/plugins/core/dataloader.py [hunk 4]
```diff
@@ -290,8 +295,6 @@ class GenericDataLoader(DataLoader):
         >>> from synthcity.plugins.core.dataloader import GenericDataLoader
         >>> X, y = load_diabetes(return_X_y=True, as_frame=True)
         >>> X["target"] = y
-        >>> # Important note: preprocessing data with OneHotEncoder or StandardScaler is not needed or recommended.
-        >>> # Synthcity handles feature encoding and standardization internally.
```
Comment (bcebere) [inline src/synthcity/plugins/core/dataloader.py:294]: Why did you remove these lines?

File: tests/metrics/test_api.py [hunk 5]
```diff
@@ -167,3 +168,61 @@ def test_weighted_metric(test_plugin: str) -> None:
         WeightedMetrics(
             [("sanity", "common_rows_proportion"), ("performance", "xgb")], [0.5, 0.5]
         )
+
+
+@pytest.mark.parametrize(
+    "fairness_column, rule, strict, add_hoc_vals",
```
Comment (bcebere) [inline tests/metrics/test_api.py:174]: Do you mean `ad_hoc`?

**Review by bcebere — CHANGES_REQUESTED** [review]
Great work!

Some changes are still needed, after that this can be merged

File: tests/metrics/test_api.py (hunk 5 above)
Comment (robsdavis) [reply tests/metrics/test_api.py]: yes, fixed

File: src/synthcity/plugins/core/dataloader.py (hunk 4 above)
Comment (robsdavis) [reply src/synthcity/plugins/core/dataloader.py]: Accident, re-instated.

File: src/synthcity/benchmark/utils.py (hunk 3 above)
Comment (robsdavis) [reply src/synthcity/benchmark/utils.py]: Deleted.
